src/Nox.Types/Types/Volume/Volume.cs

```diff
@@ -1,53 +1,167 @@
+using System.Globalization;
 using Nox.Types.Common;
 
 namespace Nox.Types;
 
 /// <summary>
 /// Represents a Nox <see cref="Volume"/> type and value object.
 /// </summary>
-public class Volume : Measurement<Volume, VolumeUnit>
+public class Volume : ValueObject<QuantityValue, Volume>
 {
+    public const ushort QuantityValueDecimalPrecision = 6;
+
+    private VolumeTypeOptions _volumeTypeOptions = new();
+    private readonly VolumeTypeUnit _unit;
+    private readonly VolumeUnit _volumeUnit = null!;
+
+    public VolumeTypeUnit Unit
+    {
+        get => _unit;
+        private init
+        {
+            _unit = value;
+            _volumeUnit = Enumeration.ParseFromName<VolumeUnit>(_unit.ToString());
+        }
+    }
+
+    /// <inheritdocs />
+    public override string ToString() => ToString(CultureInfo.InvariantCulture);
+
     /// <summary>
-    /// Creates a new instance of <see cref="Volume"/> object in cubic feet.
+    /// Returns a string representation of the <see cref="Volume"/> object using the specified <see cref="IFormatProvider"/>.
     /// </summary>
-    /// <param name="value">The value to create the <see cref="Volume"/> with</param>
-    /// <returns></returns>
-    /// <exception cref="TypeValidationException"></exception>
-    public static Volume FromCubicFeet(QuantityValue value)
-        => From(value, VolumeUnit.CubicFoot);
+    /// <param name="formatProvider">The format provider for the volume value.</param>
+    /// <returns>A string representation of the <see cref="Volume"/> object with the value formatted using the specified <see cref="IFormatProvider"/>.</returns>
+    public string ToString(IFormatProvider formatProvider) => $"{Value.ToString(formatProvider)} {_volumeUnit}";
 
     /// <summary>
     /// Creates a new instance of <see cref="Volume"/> object in cubic meters.
     /// </summary>
     /// <param name="value">The origin value to create the <see cref="Volume"/> with</param>
     /// <returns></returns>
     /// <exception cref="TypeValidationException"></exception>
-    public static Volume FromCubicMeters(QuantityValue value)
-        => From(value, VolumeUnit.CubicMeter);
+    public new static Volume From(QuantityValue value)
+        => From(value, new VolumeTypeOptions());
 
     /// <summary>
-    /// Creates a new instance of <see cref="Volume"/> object in cubic meters.
+    /// Creates a new instance of <see cref="Volume"/> object in specified unit.
     /// </summary>
-    /// <param name="value">The origin value to create the <see cref="Volume"/> with</param>
+    /// <param name="value">Value to create a <see cref="Volume"/>.</param>
+    /// <param name="unit">Unit to create a <see cref="Volume"/>.</param>
     /// <returns></returns>
     /// <exception cref="TypeValidationException"></exception>
-    public new static Volume From(QuantityValue value)
-        => From(value, VolumeUnit.CubicMeter);
+    public static Volume From(QuantityValue value, VolumeUnit unit)
```

Comment:
We don't want to expose VolumeUnit in the construction method, so please remove this one.

Reply:
Method removed.